docs/internals/data-directory.md

```diff
@@ -0,0 +1,43 @@
+# Quickwit data directory
+
+Quickwit operates on a local directory to store temporary files. By default, this working directory is named `qwdata` and placed right next to the Quickwit binary. The following is a tree of how that directory might look like after some time of operating Quickwit.
+
+```bash
+qwdata
+├── cache
+│   └── splits
+|       ├── 03BSWV41QNGY5MZV9JYAE1DCGA.split
+│       └── 01GSWV41QNGY5MZV9JYAE1DCB7.split
+├── delete_task_service
+│   └── wikipedia
+│       └── scratch
+├── indexing
+│   └── wikipedia
+│       └── _ingest-api-source
+│           └── scratch
+└── queues
+    ├── partition_id
+    ├── wal-00000000000000000028
+    └── wal-00000000000000000029
+```
+
+- `/queues`: This directory contains write ahead log files of the ingest API. When data is sent to Quickwit via the ingest API, the data is stored in these queues to guard against data lost. The indexing pipelines then consume those queues at their own pace. Whenever an indexing pipeline commits based on `commit_timeout_secs`, the queue is truncated to free up the unnecessarily occupied storage. This directory is created only if the ingest API service is running on your node.
+
+- `indexing`: This directory hold the local indexing directory of each index managed by Quickwit. The local indexing directories are further separated by indexing source. In the above tree, you can see there is an indexing folder corresponding to `_ingest-api-source` source for `wikipedia` index. This directory gets created only on nodes running the indexing service.
+
+- `delete_task_service`: This directory is used by the Janitor service to apply deletes on indexes. During this process, splits are downloaded, a new split is created while applying deletes and uploaded to the target storage. This directory gets created only on nodes running Janitor service.
+
+- `cache`: This directory is used for caching splits created on this node. As we want to reduce time-to-search, we often creates splits that are far from mature in size and number of document. This leads to new splits going through a set of merges before getting mature. In order to avoid downloading a split that already existed on disk, we keep hold of newly created splits in cache as much as possible. This cache limits on number of splits as well as the overall size of splits it holds. Splits are evicted based on age only (oldest gets evicted).
+
+## Keeping disk space in check
```

Comment:
I think this part is too long, it's fine for internal docs but we should make the user life easier on this and we should probably be more concrete and present a good rule of thumb.
Let me find some time this weekend to propose something.

Reply:
I will try to shorten it then